molecule/structure_checker2.h

```diff
@@ -0,0 +1,133 @@
+/****************************************************************************
+ * Copyright (C) from 2009 to Present EPAM Systems.
+ *
+ * This file is part of Indigo toolkit.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ***************************************************************************/
+
+#ifndef __structure_checker3__
+#define __structure_checker3__
+
+#include "base_c/defs.h"
+#include "base_cpp/exception.h"
+#include <string>
+#include <unordered_set>
+#include <vector>
+
+namespace indigo
+{
+
+    class BaseMolecule;
+    class BaseReaction;
+
+    class DLLEXPORT StructureChecker2
+    {
+    public:
+        enum CheckTypeCode
+        {
+            CHECK_NONE = 0x00000000,         // Check none
```

Comment:
duplicate code and overcomplicate logic. Instead it is worth to create two maps with 
types_map<string, CheckMessageCode>
messages_map<CheckMessageCode , string>